### PR TITLE
Exception aggregate mechanism

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description

Adds the new field according to https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md#new-mechanism-fields

I didn't yet implement exception aggregates for anything. That can be done in follow-up PRs.
The current CI failures are due to targeting the v8 branch, I believe. At least, I can't see any failures related to the new code.

## :bulb: Motivation and Context

https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md

A tiny step towards #1859

## :green_heart: How did you test it?

Updated tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

Use it for PlatformException in the Flutter code.

#skip-changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,10 @@
   - Now the device context from Android is available in `BeforeSendCallback`
 - Set ip_address to {{auto}} by default, even if sendDefaultPII is disabled ([#1665](https://github.com/getsentry/sentry-dart/pull/1665))
   - Instead use the "Prevent Storing of IP Addresses" option in the "Security & Privacy" project settings on sentry.io
+
+### Features
+
+- Add support for exception aggregates ([#xxxx](https://github.com/getsentry/sentry-dart/pull/xxxx))
```
Comment (ueman) [inline CHANGELOG.md]:
```suggestion
- Add support for exception aggregates ([#1866](https://github.com/getsentry/sentry-dart/pull/1866))
```

**Review by buenaflor — APPROVED** [review]

**Comment by buenaflor** [comment]
some tests are still failing it looks like

**Comment by ueman** [comment]
@buenaflor 
I've fixed the issues. Is the target branch still the correct one, or should I target another branch?

**Comment by buenaflor** [comment]
`release/8.0.0` is fine, thx, I'll merge once everything is green (except `sentry-dio`, that's expected to fail currently fyi)

File: dart/lib/src/protocol/mechanism.dart [hunk 2]
```diff
@@ -44,6 +44,38 @@ class Mechanism {
   /// This may be because they are created at a central place (like a crash handler), and are all called the same: Error, Segfault etc. When the flag is set, Sentry will then try to use other information (top in-app frame function) rather than exception type and value in the UI for the primary event display. This flag should be set for all "segfaults" for instance as every single error group would look very similar otherwise.
   final bool? synthetic;
 
+  /// An optional boolean value, set `true` when the exception is the exception
+  /// group type specific to the platform or language.
+  /// The default is false when omitted.
+  /// For example, exceptions of type (PlatformException)[https://api.flutter.dev/flutter/services/PlatformException-class.html]
```
Comment (ueman) [inline dart/lib/src/protocol/mechanism.dart]:
```suggestion
  /// For example, exceptions of type [PlatformException](https://api.flutter.dev/flutter/services/PlatformException-class.html)
```

**Comment by ueman** [comment]
@buenaflor Is there something I still need to do here? As far as I can tell, the remaining failing jobs aren't related to this PR.

**Comment by buenaflor** [comment]
I'll merge main into v8 soon and the ci should be green then

**Pull request merged**